shell/common/rasterizer.cc

```diff
@@ -469,7 +469,7 @@ RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
         raster_status == RasterStatus::kSkipAndRetry) {
       return raster_status;
     }
-    if (external_view_embedder != nullptr) {
+    if (external_view_embedder != nullptr && (!raster_thread_merger_ || raster_thread_merger_->IsMerged())) {
```

Comment:
Is raster_thread_merge_ thread-safe?  `raster_thread_merger_->Enable()` is called from the platform thread.

Reply:
Looks like there are checks for raster_thread_merger_ being null on the platform thread while it is created on the raster thread, whilst not being atomic.  That looks like a problem itself.

Reply:
raster_thread_merge_ is always on raster task runner. `raster_thread_merger_->IsMerged()` is thread safe.

Reply:
Looks like enabling the merger is protected by a mutex: https://github.com/gaaclarke/engine/blob/e1e90f2a2be3bbd61237cb794ec2e21ddfeace5e/fml/raster_thread_merger.cc#L101:L102

Reply:
we still need to address the `flutter_view_` race, no?

Reply:
I think @blasten is right about the race condition.  Even with the threads merged we could end up in a SubmitFrame with a null `flutter_view_` which the assert says shouldn't' happen.  FWIW I ran the code with a null value for `flutter_view_` by removing the assert and nothing crashed, so it looks like objc handled the null checks fine.  We can just double check that null is handled fine and remove that assert maybe.